Closes Feature/#14

nima opened the pull request:
Cleaned up the feature branch history, new CR (and final one hopefully).

nima committed:
Implemented indentation logic in `cpf' and `cpfi'

nima committed:
Removed a redundant/broken `cpf' script

baua committed:
Render `cpf' into a standard module

Additionally, alphabetically order the list of tests.

nima commented on module/cpf.sh:
```diff
@@ -26,17 +29,22 @@ declare -A COLORS=(
+    local module_path
+
+    if [ "${profile}" = "core" ]; then
+	    module_path=${SIMBOL_SCM?}/module
```
That indentation looks off.

nima commented on module/cpf.sh:
```diff
@@ -26,17 +29,22 @@ declare -A COLORS=(
 
 #. cpf:module -={
-function ::cpf:module_is_modified() {
+function ::cpf:module_is_unchanged() {
```
What was the reason for the name change here?

baua replied:
That it confused me. It returned 0 when it was not modified, which I interpret as true. I just believed the use of it is clearer if you see the function name as a question. I hope that makes sense.

nima replied:
Okay, so two things:
1. Your first comment can be resolved by changing it to use the variables `${TRUE}` and `${FALSE}`; those are synonymous with `${CODE_SUCCESS}` and `${CODE_FAILURE}`; but make sense in the context of a boolean function.
2. Your second comment is void in the sense that `is_modified` is as much of a question as `is_unchanged`.  So adding a change like this just confuses things.  But if it was flipped the wrong way then the change is valid.  We should start using the boolean variables for boolean functions I think is the key.

nima replied:
Also you changed the function name to "is_unchanged", but then the variable receiving this answer is called "amended".  That is very confusing.

nima replied:
I cleaned this up and pushed - have a look and let me know if it makes sense to you?

nima commented on module/cpf.sh:
```diff
@@ -26,17 +29,22 @@ declare -A COLORS=(
-        [ ${PIPESTATUS[0]} -ne 0 ] || e=${amended}
+    local module_path
+
+    if [ "${profile}" = "core" ]; then
```
I'm a bit worried about this, it's confusing since the "profile" and `SIMBOL_PROFILE` are what we use to determine the context under which simbol is operating; for example it would be set to `DEFAULT` by default, but then the user creates profiles for the context pertaining to their job.  So at AWS I have my profile set to `AWS`, and when I was at Sirca it was set to `SIRCA`.  The new term "core" is out-of-place here.

nima reviewed (commented):
Let's make some fixes in here now and get this branch out into develop!

nima committed:
Moved repetitive module logic code to libsimbol

baua commented on lib/libsh/libsimbol/libsimbol.sh:
```diff
@@ -351,6 +354,50 @@ function core:softimport() {
+    if [ $# -eq 1 ]; then
+        local module="$1"
+
+        if [ -e ${SIMBOL_CORE_MOD?}/${module//\./\/}.sh ]; then
```
Did you consider a case statement? Maybe later on, we want to put in a warning if a module with the same name exists in `CORE` and `USER`.

baua reviewed (approved):
thumbs up if both of my comments are of now concern.

nima committed:
Cleaned up boolean variables and fixed output

Output color has been simplified: the printout of the function/method
now re-uses the module printout functions instead of re-implimenting
it.

Using the boolean TRUE and FALSE variables for booleans, not zeros and
ones.

nima commented:
Final thing - the `assert` ids in the static tests, have a look at `module/hgd.sh` - those are much easier to track down on failure.  For example `hgd.resolve.1.3` vs `1.1`.

baua committed:
Updated unit tests to address last set of changes
